packages/core/src/components/DataTable/index.tsx

```diff
@@ -403,40 +459,39 @@ export class DataTable extends React.Component<DataTableProps & WithStylesProps,
 
     return (
       <AutoSizer disableHeight={!autoHeight}>
-        {({ height, width }: { height: number; width: number }) => (
-          <>
-            {this.shouldRenderTableHeader() && this.renderTableHeader(width)}
-            <div className={cx(styles.table_container, { width })}>
-              <Table
-                ref={propagateRef}
-                rowGetter={this.rowGetter(expandedData)}
-                headerHeight={this.getColumnHeaderHeight()}
-                height={
-                  autoHeight
-                    ? height -
-                      (this.shouldRenderTableHeader() ? getHeight(rowHeight, tableHeaderHeight) : 0)
-                    : this.getTableHeight(expandedData)
-                }
-                rowCount={expandedData.length}
-                rowHeight={HEIGHT_TO_PX[rowHeight!]}
-                width={this.props.width || width}
-                rowStyle={this.getRowStyle(expandedData)}
-                sort={this.sort}
-                sortBy={sortBy}
-                sortDirection={sortDirection}
-                headerRowRenderer={ColumnLabels(this.props)}
-                onRowClick={this.handleRowClick}
-              >
-                {expandable && renderExpandableColumn(cx, styles, expandedRows, this.expandRow)}
-
-                {selectable &&
-                  renderSelectableColumn(selectedRows, this.handleSelection, expandable)}
-
-                {renderDataColumns(this.keys, editMode, this.onEdit, this.props)}
-              </Table>
-            </div>
-          </>
-        )}
+        {({ height, width }: { height: number; width: number }) => {
+          const tableHeight = autoHeight
+            ? height -
+              (this.shouldRenderTableHeader() ? getHeight(rowHeight, tableHeaderHeight) : 0)
+            : this.getTableHeight(expandedData);
+          return (
+            <>
+              {this.shouldRenderTableHeader() && this.renderTableHeader(width)}
+              <div className={cx(styles.table_container, { width })}>
+                <Table
+                  height={tableHeight}
+                  width={this.props.width || width}
+                  rowCount={expandedData.length}
+                  rowGetter={this.rowGetter(expandedData)}
+                  sort={this.sort}
+                  sortBy={sortBy}
+                  sortDirection={sortDirection}
+                  headerHeight={this.getColumnHeaderHeight()}
+                  headerRowRenderer={ColumnLabels(this.props)}
+                  rowHeight={dynamicRowHeight ? this.cache.rowHeight : HEIGHT_TO_PX[rowHeight!]}
+                  rowStyle={this.getRowStyle(expandedData)}
+                  overscanRowCount={100}
```

Comment:
do we need this big of an overscan?

Reply:
Good catch, we can't use `showAllRows` with `dynamicRowHeight` without this, but it's unnecessary otherwise. Changing to
`overscanRowCount={dynamicRowHeight && showAllRows ? expandedData.length : 2}`

Reply:
Maybe overscan be the number of rows currently displayed? Or maybe half?

Reply:
This is possible, but pretty tricky to implement with `dynamicRowHeight`, not sure it's necessary normally since there used to be 0 overscan

Reply:
Yeah no worries then.